handsontable/src/plugins/filters/conditionCollection.js

```diff
@@ -46,6 +53,8 @@ class ConditionCollection {
     } else {
       this.filteringStates.init(this.hot.columnIndexMapper.getNumberOfIndexes());
     }
+
+    this.hot.addHook('afterFilter', () => this.#onAfterFilter());
```

Comment:
```suggestion
```
If possible, I would keep this internal module as clean as possible... meaning I would not use a plugin's hook but instead call the ".setPreviousConditionStack(null)" method manually from [here](https://github.com/handsontable/handsontable/blob/feature/dev-issue-910/handsontable/src/plugins/filters/filters.js#L555). I would avoid creating more dependencies between the module and the "hot" instance as this creates later too tied modules difficult in refactor. 

In the best case, the module should only accept the "columnIndexMapper" module and the "getCellMeta" function getter instead of the whole "hot" instance.